core/src/main/java/org/apache/gravitino/metrics/source/RelationDatasourceMetricsSource.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.metrics.source;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import org.apache.commons.dbcp2.BasicDataSource;
+
+public class RelationDatasourceMetricsSource extends MetricsSource {
+
+  public RelationDatasourceMetricsSource(BasicDataSource dataSource) {
+    super(MetricsSource.RELATION_DATASOURCE_METRIC_NAME);
+    MetricRegistry metricRegistry = getMetricRegistry();
+    metricRegistry.register("activeConnectionCount", (Gauge<Integer>) dataSource::getNumActive);
+    metricRegistry.register("idleConnectionCount", (Gauge<Integer>) dataSource::getNumIdle);
+    metricRegistry.register("maxConnectionCount", (Gauge<Integer>) dataSource::getMaxTotal);
```

Comment:
could you add test in `TestExtractMetricNameAndLabel.testMapperConfig` like `jvm.total.used`?

Reply:
similarly, we are supposed to have 'idleConnections', 'maxConnections'.